datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -97,6 +102,27 @@ pub fn name_to_op(name: &str) -> Result<Operator> {
     }
 }
 
+fn name_to_scalar_function(name: &str) -> Result<BuiltinScalarFunction> {
+    match name {
```

Comment:
I wonder if you can use `FromStr` for this: https://github.com/apache/arrow-datafusion/blob/2bbc5063fefcc51ca98300e74b1bbc19ef4827ad/datafusion/expr/src/built_in_function.rs#L445-L458

So like 
```rust
let func = BuiltinScalarFunction::from_str(name)?;
```

Or perhaps you need a different list of names to match what is in substrait 🤔